### ISSUE TITLE
Add npm/module support

### ISSUE THREAD

**Issue by kylehalleman** [umbrellajs/umbrella#67]
Hi,

Love this project, just came across it the other day. Right now I'm installing via bower and adding module.exports manually so I can easily pull in umbrella into my existing code (using ES modules and webpack). It'd be awesome if this was built into the core and released on npm.

Keep up the good work!

**Comment by franciscop** [comment]
Can you show me how you are doing it, please? Also how do you deal with `ajax()` and `parseJson()` being separated functions, just ignoring them? (note, there is *also* a method called `.ajax()`, but I mean the plugin fn.ajax)

**Comment by kylehalleman** [comment]
I'm doing it the lazy right now, because I only need to support my own project, so at the bottom of the file I'm just adding

```
module.exports = {
  u: u,
  ajax: ajax,
};
```

(not using `parseJson` so I'm not exporting it)

However, to implement this in a more compatible way for other people (using AMD, CommonJS, etc), you'd need to add more code. You might want to look at [this repo](https://github.com/umdjs/umd) for samples of different module exporting code (I think that `returnExportsGlobal.js` might be what you want in this case).

**Comment by franciscop** [comment]
`parseJson()` is not supposed to be used by anyone (only internal modules) so we're good.

The problem with `returnExportsGlobal.js` is that it requires to wrap ALL of the code into braces, which is tricky with the current build system. However your solution looks great, maybe with a simple check it could be improved:

```js
if (module && module.exports) {
  module.exports = {
    u: u,
    ajax: ajax
  };
}
```

AMD and CommonJS [seems to be dead already](https://www.reddit.com/r/javascript/comments/3vqgai/eli5_commonjs_vs_amd_vs_es6/), and `ES6's export` is still not supported by anything.

**Comment by franciscop** [comment]
Do you mind adding a small explanation in the README.md please? I have just implemented it, but I am not sure when/how Webpack is useful so you would do a much better job explaining it. You can just use the Github interface if you don't want to go all the drill of fork+clone+commit+push. Also if you have any problem with the implementation please let me know.

**Comment by kylehalleman** [comment]
Updated the README in #68.

I know your current build system doesn't support wrapping the library in an IIFE, you might want to explore building the library using some sort of ES modules in the future so down the line people could pull in individual functions so the library can be even smaller when not everything is needed.

I would also suggest putting this in the npm registry so it's easier to grab and pull into a project. I haven't done it myself before, but [this page](https://docs.npmjs.com/getting-started/publishing-npm-packages) makes it seem straightforward. I'd offer to do it as a PR, but I think as the repo owner you'd have to/should do it.

**Comment by franciscop** [comment]
I think under 3kb, the library is small enough that it doesn't justify a custom build system for cherry picking plugins. Developers *can* do it however, but there are also inter-dependencies within the methods which makes it difficult.

I thought it was on NPM but I just checked and not, it's not. That's an oversight, I'll fix it ASAP. Also thanks for the PR.

**Comment by kylehalleman** [comment]
Haha definitely wasn't thinking when I mentioned modularized builds (forgot how tiny this library is).

**Comment by franciscop** [comment]
It is now published as `umbrellajs` in NPM registry, does it work for you?

**Comment by kylehalleman** [comment]
Yep! Thanks a lot!
